tests/test_scurl.py

```diff
@@ -30,5 +30,46 @@ def test_check_invalid_urls(self):
                                  stdlib.urljoin(invalid_url, invalid_url_2))
 
 
+    def test_urljoin_no_canonicalize(self):
+        bases = [
+            'http://example.com',
+            'http://example.com/',
+            'http://example.com/white space',
+            'http://example.com/white space/foo bar/',
+            'file://example.com/white space',
+            'file://example.com/'
+        ]
+        urls = [
+            '',
+            'foo/bar',
+            'white space',
+            'white space/foo bar',
+            'http://example2.com/',
+            'http://example2.com',
+            'http://example2.com/white space',
+            'file://a/b/c/d'
+        ]
+
+        for base in bases:
+            for url in urls:
+                self.assertEqual(scurl.urljoin(base, url),
+                                 stdlib.urljoin(base, url))
+                self.assertEqual(scurl.urljoin(url, base),
+                                 stdlib.urljoin(url, base))
+
+    def test_urljoin_invalid_host(self):
+        bases = [
+            'http://example].com',
+            'http://[example.com'
+        ]
+        urls = [
+            'http://[example.com/foo/bar',
+            'http://example.com]/foo/bar'
+        ]
+
+        for base in bases:
+            for url in urls:
+                self.assertRaises(ValueError)
```

Comment:
Probably you wanted to check if some method raises ValueError?

Reply:
@lopuhin Do you mean I should add some urls that don't raise the ValueError? 😄

Reply:
@nctl144 I mean that as it stands, this test checks nothing I think (``base`` and ``url` are not used), this is how ``assertRaises`` is meant to be used: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises

Reply:
Oh rightttt... I don't even know why I added this to the test... I guess it was because I was stressing out while trying to fix the Scrapy PR. I will fix this now!